app/controllers/api/v1/customers_controller.rb

```diff
@@ -112,11 +113,11 @@ def transform_address!(attributes, from, to)
         end
 
         if address[:state_name].present?
-          address[:state] = Spree::State.find_by(name: address[:state_name])
+          address[:state] = Spree::State.find_by(name: address[:state_name][:name])
```

Comment:
This reads a bit funny. I think that we should just use `state` instead of `state_name` now.